### ISSUE TITLE
metrics computation

### ISSUE THREAD

**Issue by DukeofCambridge** [dvlab-research/Ref-NPR#9]
Hi, I wonder how did you compute your robustness metric and Ref-LPIPS as mentioned in Figure B.1.
I noticed you compute PSNR, SSIM and LPIPS in calc_metric.py but that seems to be computed on stylized view and original view images. Did you provide codes that work for what you mentioned in Figure B.1. and Table 1? Thanks very much.

**Comment by JulianJuaner** [comment]
Hi, sorry for the late reply.

For robustness, one simple implementation is to select some rendered samples from exp_A, then optimize several NPR models B, C, D... And for each model, compute the PSNR value of the rendered frames.
```python
A = "exp_A/to/test_renders_path"
B = "exp_B/to/test_renders_path"
psnr = []
paths = sorted([m for m in sorted(os.listdir(A)) if 'png' in m])
for path in paths:
    im_a = torch.from_numpy(cv2.resize(cv2.imread(os.path.join(A, path)), (400, 400)))/255
    im_b = torch.from_numpy(cv2.resize(cv2.imread(os.path.join(B, path)), (400, 400)))/255
    mse = torch.nn.functional.mse_loss(im_a, im_b)
    psnr += [-10.0 * math.log10(mse.detach().item())]
# here we compute the top 100 frames
print(sum(sorted(psnr)[-100:])/100)
```

For LPIPS, here is a sample code sample to compare in one experiment. You may need to modify the exp name and out_dir
```python
import lpips
import imageio
from tqdm import tqdm
import torch.nn.functional as F
import torch
import cv2
import os
import numpy as np
from icecream import ic

device = "cuda"
loss_fn_vgg = lpips.LPIPS(net='vgg').eval().to(device) 
# exps = ["01_ABC_mic", "03_colorful_flower", "02_vangoh_playground", "10_prestyle_horns", "05b_sketch_hotdog_xf",
#         "04_leaf_chair", "11a_ficus_01", "13_truck_style"]

exps = ["03_colorful_flower"]

for exp in exps:
    scale = 0.25 # rescale for normal input size of VGG (256x256).
    static_style = "./exps/g6_ds_weak_color/" + exp + "/exp_out/style_image_0.png"
    out_dir = "../Few-Shot-Patch-Based-Training/data/" + exp + "/res__P/0030000"
    
    out_imgs = [
        torch.from_numpy(cv2.cvtColor(cv2.imread(os.path.join(out_dir, m), 1), cv2.COLOR_BGR2RGB).astype(np.float32)/255.0).permute(2,0,1).to(device=device) for m in sorted(os.listdir(out_dir))
    ]
    style = cv2.cvtColor(cv2.imread(static_style, 1), cv2.COLOR_BGR2RGB).astype(np.float32)/255.0
    style = torch.from_numpy(style).permute(2,0,1).to(device=device)
    style = F.interpolate(style.unsqueeze(0), size=None,
                                    scale_factor=scale,
                                    mode="bilinear")[0]
    static_list = []
    for i in tqdm(range(len(out_imgs))):
        img_i = F.interpolate(out_imgs[i].unsqueeze(0), size=None,
                                    scale_factor=scale,
                                    mode="bilinear")[0]
        style = F.interpolate(style.unsqueeze(0), size=img_i.shape[-2:],
                                    mode="bilinear")[0]
        static_list += [loss_fn_vgg(style.contiguous(),
                        img_i.contiguous(), normalize=True).item()]

    ic(sum(sorted(static_list)[:-10])/10)
```

**Comment by DukeofCambridge** [comment]
Thanks so much!
